js_modules/dagster-ui/packages/ui-core/src/assets/LaunchAssetExecutionButton.tsx

```diff
@@ -269,93 +278,117 @@ export const LaunchAssetExecutionButton = ({
+          {iconOnly ? (
+            <UnstyledButton
+              style={{padding: 4}}
+              onClick={(e) => firstAction(firstOption.assetKeys, e)}
+              disabled={!firstOption.assetKeys.length}
+            >
+              {firstOption.icon}
+            </UnstyledButton>
+          ) : (
+            <MaterializeButton
+              intent={primary ? 'primary' : undefined}
+              data-testid={testId('materialize-button')}
+              onClick={(e) => firstAction(firstOption.assetKeys, e)}
+              style={{
+                borderTopRightRadius: 0,
+                borderBottomRightRadius: 0,
+                borderRight: `1px solid rgba(255,255,255,0.2)`,
+              }}
+              disabled={!firstOption.assetKeys.length}
+              icon={
+                loading ? (
+                  <Spinner purpose="body-text" />
+                ) : (
+                  <Icon
+                    name={
+                      featureEnabled(FeatureFlag.flagUseNewObserveUIs)
+                        ? 'execute'
+                        : 'materialization'
+                    }
+                  />
+                )
+              }
+            >
+              {firstOption.label}
+            </MaterializeButton>
+          )}
         </Tooltip>
 
-        <Popover
-          isOpen={isOpen}
-          onInteraction={(nextOpen) => setIsOpen(nextOpen)}
-          position="bottom-right"
-          content={
-            <Menu>
-              <Tooltip
-                canShow={!!secondOption.disabledReason}
-                content={secondOption.disabledReason || ''}
-                useDisabledButtonTooltipFix
-                position="left"
-              >
-                <MenuItem
-                  key={secondOption.label}
-                  text={secondOption.label}
-                  icon={secondOption.icon}
-                  data-testid={testId('materialize-secondary-option')}
-                  disabled={secondOption.assetKeys.length === 0}
-                  onClick={(e) => secondAction(secondOption.assetKeys, e)}
-                />
-              </Tooltip>
-              {showChangedAndMissingOption ? (
-                <MenuItem
-                  text="Materialize unsynced"
-                  icon="changes_present"
-                  disabled={!!materializeOption.disabledReason}
-                  onClick={() => setShowCalculatingUnsyncedDialog(true)}
-                />
-              ) : null}
-              <MenuItem
-                text="Open launchpad"
-                icon="open_in_new"
-                onClick={(e: React.MouseEvent<any>) => {
-                  materialize.onClick(firstOption.assetKeys, e, true);
-                }}
-              />
-              {additionalDropdownOptions?.map((option) => {
-                if (!('label' in option)) {
-                  return option;
-                }
-
-                const item = (
+        {iconOnly ? null : (
```

Comment:
same this is the only change (wrapping in a ternary), everything else is whitepsace